app/jobs/algolia_reindex_job.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Background job to reindex all relevant database records in Algolia.
+#
+# Usage:
+#
+# To send job to a background worker to execute:
+#
+# > AlgoliaReindexJob.perform_async
+#
+# To execute in-line immediately from Rails console:
+#
+# > AlgoliaReindexJob.new.perform
+#
+class AlgoliaReindexJob
+  def self.perform_async
+    Delayed::Job.enqueue(AlgoliaReindexJob.new)
+  end
+
+  def perform
+    Resource.reindex
+    Service.reindex
```

Comment:
This is going to logically conflict with @jjfreund's changes in https://github.com/ShelterTechSF/askdarcel-api/pull/367, where he's slightly modifying the reindexing commands.

I think we should pull the logic into a common place where both the Rake task and this delayed job call the same function.

Reply:
His change is only to reindex approved services, right? I don't see a conflict there.

Reply:
Richard is right, Cliff should use the same business logic I do (I believe). There are two things that seem wrong here -- first, the where clauses do not restrict the results to active statuses (see my commit for reference). Second, since the services and resources share an index, the second statement has to use the reindex! method, or else it will blow away the records created by the first index.

Reply:
@jjfreund Thanks! I've pushed a change to make it so that the AlgoliaReindexJob uses the following commands. I also changed the `algolia.rake` script to use this common code. Any thoughts?
```Ruby
    Resource.where(status: :approved).reindex
    # Since Service and Resource share an algolia index, we use `reindex!` so
    # Service reindexing does not clear out Resource index records.
    Service.where(status: :approved).reindex!
```